components/schema-migrator/migrations/director/20210915134043_add_providers_to_views.up.sql

```diff
@@ -0,0 +1,325 @@
+BEGIN;
+
+CREATE OR REPLACE FUNCTION apps_subaccounts_func()
+    RETURNS TABLE
+            (
+                id                 uuid,
+                tenant_id          uuid,
+                provider_tenant_id uuid
+            )
+    LANGUAGE plpgsql
+AS
+$$
+BEGIN
+    RETURN QUERY
+        SELECT l.app_id                 AS id,
+               asa.selector_value::uuid AS tenant_id,
+               asa.selector_value::uuid AS provider_tenant_id
+        FROM labels l
+                 -- 2) Get subaccounts in those scenarios (Putting a subaccount in a
+                 -- scenario will reflect on creating an ASA for the subaccount.
+                 JOIN automatic_scenario_assignments asa
+                      ON asa.tenant_id = l.tenant_id AND l.value ? asa.scenario::text AND
+                         asa.selector_key::text = 'global_subaccount_id'::text
+             -- 1) Get all scenario labels for applications
+        WHERE l.app_id IS NOT NULL
+          AND l.key::text = 'scenarios'::text;
+END
+$$;
+
+CREATE OR REPLACE FUNCTION consumers_provider_for_runtimes_func()
+    RETURNS TABLE
+            (
+                provider_tenant  jsonb,
+                consumer_tenants jsonb
+            )
+    LANGUAGE plpgsql
+AS
+$$
+BEGIN
+    RETURN QUERY
+        SELECT l1.value -> 0 AS provider_tenant, l2.value AS consumer_tenants
```

Comment:
You can use `l1.value ->> 0` to extract text and do not cast below.

Reply:
Also, keep in mind that our data (especially on dev) is not that consistent.

There are `global_subaccount_id` labels with empty values or even worse with values that are not UUIDs. This will make the cast below fail.